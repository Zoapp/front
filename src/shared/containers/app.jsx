/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import Zrmc, {
  Button, Content, Fab, Snackbar, Tabbar, Tab,
  Toolbar, ToolbarRow, ToolbarSection, ToolbarTitle, ToolbarIcon,
  Drawer, DrawerHeader, DrawerContent,
} from "zrmc";
import PropTypes from "prop-types";
import { Link, Route, Switch, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import Screen from "./screen";
import UserBox from "./userBox";
import { appSetTitle, appRemoveLastMessage } from "../actions/app";
import { initAuthSettings } from "../actions/initialize";
import { apiAdminRequest } from "../actions/api";

class App extends React.Component {
  constructor(props) {
    super(props);
    const { type: drawer, above: aboveToolbar, themeDark: drawerThemeDark } = props.design.drawer;
    const snackbar = this.props.message;
    this.state = {
      needUpdate: true,
      activeTab: props.activeTab,
      drawer,
      drawerOpen: false,
      drawerThemeDark,
      aboveToolbar,
      snackbar,
    };
  }

  componentDidMount() {
    this.props.initAuthSettings();
    this.updateAdmin();
    Zrmc.init(this, { typography: true });
  }

  componentWillReceiveProps(nextProps) {
    let snackbar = nextProps.message;
    if ((!snackbar) && nextProps.error) {
      snackbar = { message: nextProps.error };
    }
    this.setState({ snackbar });
  }

  componentDidUpdate() {
    this.updateAdmin();
  }

  onMenuClick = (event) => {
    event.preventDefault();
    this.toggleDrawer();
  }

  updateAdmin() {
    if (!this.props.isSignedIn) {
      if (!this.state.needUpdate) {
        this.setState({ needUpdate: true });
      }
      return;
    }
    if (this.state.needUpdate) {
      this.setState({ needUpdate: false });
      this.props.apiAdminRequest();
    }
  }

  handleTimeoutError() {
    this.todo = {};
  }

  handleToolbarTabChange = (name, index) => {
    this.setState({ activeTab: index });
  }

  handleDisplayScreen = () => {
    this.setState({ activeTab: 0 });
  }

  toggleDrawer = () => {
    const open = !this.state.drawerOpen;
    this.setState({ drawerOpen: open });
  }

  render() {
    let icon;
    if (this.state.drawer !== "permanent") {
      icon = <ToolbarIcon name="menu" onClick={this.onMenuClick} />;
    }
    let { isLoading } = this.props;
    if ((!isLoading) && (!this.props.admin) && this.props.isSignedIn) {
      isLoading = true;
    }
    const items = [];
    const routes = [];
    const { screens, titleName, appName } = this.props;
    let currentScreen = null;
    screens.forEach((screen) => {
      if (screen.isDrawerItem && (screen.access === "all" ||
      (this.props.isSignedIn && screen.access === "auth") ||
      ((!this.props.isSignedIn) && screen.access === "public"))) {
        let activated = false;
        if (titleName === screen.name) {
          activated = true;
          currentScreen = screen;
        }
        items.push({
          activated, ...screen,
        });
      }
      if (screen.path) {
        if (screen.path === "*" || screen.path === "/") {
          routes.push({ ...screen });
        } else {
          routes.splice(0, 0, { ...screen });
        }
      }
    });
    let tabbar;
    let toolbox;
    let fab;
    if (currentScreen) {
      if (currentScreen.panels) {
        const ac = "var(--mdc-theme-text-primary-on-primary, white)";
        const c = "rgba(255, 255, 255, 0.54)";
        tabbar = (
          <ToolbarSection>
            <Tabbar
              onChange={this.handleToolbarTabChange}
              activeTab={this.state.activeTab}
              color={c}
              activeColor={ac}
            >
              {currentScreen.panels.map((p, index) => {
                const k = `t_${index}`;
                return (
                  <Tab
                    key={k}
                    style={{ minWidth: "80px", width: "80px" }}
                    ripple
                  >
                    {p}
                  </Tab>);
              })}
            </Tabbar>
          </ToolbarSection>);
      }
      if (currentScreen.toolbox) {
        toolbox = (
          <ToolbarSection align="end" >
            {currentScreen.toolbox.map((p, index) => {
              const k = `tb_${index}`;
              return (
                <Button
                  key={k}
                  raised
                  style={{ marginRight: "48px", backgroundColor: "var(--mdc-theme-secondary, #018786)" }}
                  onClick={(e) => {
                    e.preventDefault();
                    p.onAction(p);
                  }}
                >
                  {p.title}
                </Button>);
            })}
          </ToolbarSection>);
      }
      if (currentScreen.fab) {
        fab = <Fab icon={currentScreen.fab.icon} onClick={currentScreen.fab.onAction} />;
      }
    }
    let snackbar;
    if (this.state.snackbar) {
      snackbar = (
        <Snackbar
          message={this.state.snackbar.message}
          onTimeout={() => { this.props.appRemoveLastMessage(); }}
        />);
    }
    return (
      <Content>
        <Toolbar fixed>
          <ToolbarRow>
            <ToolbarSection align="start" >
              {icon}
              <ToolbarTitle>
                <span style={{ fontWeight: "900" }}>{appName} / </span>
                <span style={{ color: "#ddd" }}>{titleName}</span>
              </ToolbarTitle>
            </ToolbarSection>
            {tabbar}
            {toolbox}
            <UserBox store={this.props.store} />
          </ToolbarRow>
        </Toolbar>
        <Drawer
          type={this.state.drawer}
          open={this.state.drawerOpen}
          above={this.state.aboveToolbar}
          onClose={this.toggleDrawer}
          themeDark={this.state.drawerThemeDark}
        >
          <DrawerHeader
            style={{
              backgroundColor: "var(--mdc-theme-secondary-dark, #004040)",
              color: "var(--mdc-theme-text-primary-on-primary, white)",
            }}
          >
            {appName}
          </DrawerHeader>
          <DrawerContent list>
            {items.map(item => (
              <Link
                key={item.id}
                href={`#${item.name}`}
                onClick={this.handleDisplayScreen}
                to={item.path}
                activated={item.activated}
                icon={item.icon}
              >{item.name}
              </Link>))}
          </DrawerContent>
        </Drawer>
        <Content>
          <Switch>
            {routes.map(screen => (
              <Route
                key={screen.id}
                path={screen.path}
                render={(props) => {
                  if (screen.render) {
                    return screen.render({ ...props, screen, activeTab: this.state.activeTab });
                  }
                  return <Screen screen={screen}>{screen.name}</Screen>;
                }}
              />
            ))}
          </Switch>
        </Content>
        {fab}
        {snackbar}
      </Content>
    );
  }
}

App.defaultProps = {
  error: null,
  message: null,
  admin: null,
  appName: "",
  screens: [],
  design: { drawer: { type: "permanent" } },
  activeTab: 0,
};

App.propTypes = {
  store: PropTypes.shape({}).isRequired,
  isSignedIn: PropTypes.bool.isRequired,
  titleName: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  error: PropTypes.shape({}),
  message: PropTypes.shape({}),
  admin: PropTypes.shape({}),
  appName: PropTypes.string,
  screens: PropTypes.arrayOf(PropTypes.shape({})),
  design: PropTypes.shape({
    drawer: PropTypes.shape({ type: PropTypes.string }),
  }),
  initAuthSettings: PropTypes.func.isRequired,
  /* appSetTitle: PropTypes.func.isRequired, */
  apiAdminRequest: PropTypes.func.isRequired,
  appRemoveLastMessage: PropTypes.func.isRequired,
  activeTab: PropTypes.number,
};

const mapStateToProps = (state) => {
  const {
    admin, screens, name, design,
  } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = (state.app && state.app.loading) ||
    (state.auth && state.auth.loading) || (state.user && state.user.loading);
  const titleName = state.app.titleName ? state.app.titleName : "";
  let error = null;
  if (state.app && state.app.error) {
    ({ error } = state.app);
  } else if (state.auth && state.auth.error) {
    ({ error } = state.auth);
  } else if ((!error) && state.user && state.user.error) {
    ({ error } = state.user);
  }
  const { message } = state.app;
  return {
    admin, isLoading, isSignedIn, titleName, error, screens, appName: name, design, message,
  };
};

const mapDispatchToProps = dispatch => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
  appRemoveLastMessage: () => {
    dispatch(appRemoveLastMessage());
  },
  initAuthSettings: () => {
    dispatch(initAuthSettings());
  },
  apiAdminRequest: () => {
    dispatch(apiAdminRequest());
  },
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
