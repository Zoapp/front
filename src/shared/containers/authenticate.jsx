/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import Zrmc, {
  Button,
  Dialog,
  DialogBody,
  DialogFooter,
  LinearProgress,
} from "zrmc";
import { connect } from "react-redux";
import {
  signIn,
  signUp,
  lostPassword,
  loadInitialState,
} from "../actions/auth";
import SignIn from "../components/auth/signIn";
import SignUp from "../components/auth/signUp";
import LostPassword from "../components/auth/lostPassword";

export class AuthenticateBase extends Component {
  constructor(props) {
    super(props);

    // accept is set to true if policyUrl is unDefined
    this.state = {
      username: "",
      email: "",
      password: "",
      display: props.display || "",
      error: null,
      displayedError: false,
      isLoading: false,
      accept: !props.policyUrl,
    };
  }

  static getDerivedStateFromProps(props, state) {
    const { error } = props;

    if (error !== state.error) {
      return {
        error,
        displayedError: false,
      };
    }
    return null;
  }

  handleCloseDialog = () => {
    Zrmc.closeDialog();
    if (this.props.onClosed) {
      this.props.onClosed();
    }
  };

  componentDidUpdate() {
    // WIP
    if (this.state.isLoading && !this.props.isLoading && !this.props.error) {
      this.handleCloseDialog();
    }
  }

  componentWillUnmount() {
    this.props.loadInitialState();
  }

  handleSignIn = (e) => {
    e.preventDefault();
    const { provider } = this.props;
    const { username, password } = this.state;
    if (username !== "" && password !== "") {
      this.setState({ isLoading: true });
      this.props.signIn(provider, username, password);
    }
  };

  handleSignUp = (e) => {
    e.preventDefault();
    const { provider } = this.props;
    const { username, email, password, accept } = this.state;
    if (username !== "" && password !== "" && email !== "") {
      this.setState({ isLoading: true });
      this.props.signUp(provider, username, email, password, accept);
    }
  };

  handleLostPassword = (e) => {
    e.preventDefault();
    const { provider } = this.props;
    const { email } = this.state;
    if (email !== "") {
      this.setState({ isLoading: true });
      this.props.lostPassword(provider, email);
    }
  };

  handleRecoverPassword = (e) => {
    e.preventDefault();
    // TODO
  };

  switchDisplay = (
    display = this.state.display === "signin" ? "signup" : "signin",
  ) => {
    if (this.state.display !== display) {
      this.setState({
        display,
        displayedError: true,
      });
    }
  };

  displaySignUp = () => {
    this.switchDisplay("signup");
  };

  displaySignIn = () => {
    this.switchDisplay("signin");
  };

  displayLostPassword = () => {
    this.switchDisplay("lostpassword");
  };

  createChangeHandler = (field) => (e) => {
    this.setState({ [field]: e.target.value });
  };

  getLostPasswordForm = (email, password, isLoading, errorMessage) => ({
    handler: this.handleLostPassword,
    form: (
      <LostPassword
        email={email}
        password={password}
        createChangeHandler={this.createChangeHandler}
        container={DialogBody}
        disabled={isLoading}
      >
        {errorMessage}
      </LostPassword>
    ),
    title: "Change your password",
    action: "Send",
    actions: [
      <Button
        key="but_cancel_pass"
        onClick={this.displaySignIn}
        className="authenticate_lostPassword"
        disabled={isLoading}
      >
        Cancel
      </Button>,
    ],
  });

  getSignUpForm = (username, email, password, isLoading, errorMessage) => ({
    handler: this.handleSignUp,
    form: (
      <SignUp
        username={username}
        email={email}
        password={password}
        createChangeHandler={this.createChangeHandler}
        container={DialogBody}
        signIn={this.displaySignIn}
        disabled={isLoading}
        policyUrl={this.props.policyUrl}
      >
        {errorMessage}
      </SignUp>
    ),
    title: "Create an account",
    action: "Register",
    actions: [],
  });

  getSingInForm = (
    username,
    password,
    isLoading,
    errorMessage,
    recoverPassword,
    signUpValidation,
  ) => {
    const actions = [];
    if (recoverPassword) {
      actions.push(
        <Button
          key="but_lost_pass"
          onClick={this.displayLostPassword}
          className="authenticate_lostPassword"
          disabled={isLoading}
        >
          Lost password ?
        </Button>,
      );
    }

    return {
      handler: this.handleSignIn,
      form: (
        <SignIn
          username={username}
          password={password}
          createChangeHandler={this.createChangeHandler}
          container={DialogBody}
          signUp={signUpValidation ? this.displaySignUp : undefined}
          disabled={isLoading}
        >
          {errorMessage}
        </SignIn>
      ),
      title: "Please login to continue",
      action: "Sign in",
      actions,
    };
  };

  render() {
    const { username, email, password, error, display } = this.state;
    const { isLoading, recoverPassword, signUpValidation, id } = this.props;

    let errorMessage = "";
    if (error && !this.state.displayedError && !isLoading) {
      errorMessage = <div className="authenticate_error">{error}</div>;
    } else if (isLoading) {
      errorMessage = <LinearProgress buffer={0} indeterminate />;
    }

    let buildedForm;
    if (display === "lostpassword") {
      buildedForm = this.getLostPasswordForm(
        email,
        password,
        isLoading,
        errorMessage,
      );
    } else if (display === "signup") {
      buildedForm = this.getSignUpForm(
        username,
        email,
        password,
        isLoading,
        errorMessage,
      );
    } else {
      buildedForm = this.getSingInForm(
        username,
        password,
        isLoading,
        errorMessage,
        recoverPassword,
        signUpValidation,
      );
    }

    buildedForm.actions.push(
      <Button
        key="but_sign"
        type="submit"
        className="authenticate_submit"
        raised
        dense
        disabled={isLoading}
      >
        {isLoading ? "Processing..." : buildedForm.action}
      </Button>,
    );

    return (
      <form id="signin-dialog-form" onSubmit={buildedForm.handler}>
        <Dialog
          id={id}
          onClose={this.handleCloseDialog}
          header={buildedForm.title}
          width="320px"
        >
          {buildedForm.form}
          <DialogFooter>{buildedForm.actions}</DialogFooter>
        </Dialog>
      </form>
    );
  }
}

AuthenticateBase.propTypes = {
  id: PropTypes.string,
  provider: PropTypes.string,
  loadInitialState: PropTypes.func.isRequired,
  signIn: PropTypes.func.isRequired,
  signUp: PropTypes.func.isRequired,
  lostPassword: PropTypes.func.isRequired,
  onClosed: PropTypes.func,
  signUpValidation: PropTypes.string,
  policyUrl: PropTypes.string,
  display: PropTypes.string,
  recoverPassword: PropTypes.bool,
  isLoading: PropTypes.bool,
  error: PropTypes.string,
};

AuthenticateBase.defaultProps = {
  provider: null,
};

const mapStateToProps = (state) => {
  const { error, loading } = state.auth;
  const { configuration } = state.app;

  let signUpValidation;
  if (configuration.backend.auth.signUp) {
    signUpValidation = configuration.backend.auth.signUp.validation;
  }

  const recoverPassword = !!configuration.backend.auth.recoverPassword;
  const { policyUrl } = configuration.backend.auth;

  return {
    signUpValidation,
    recoverPassword,
    policyUrl,
    isLoading: loading,
    error: error && error.message,
  };
};

const mapDispatchToProps = (dispatch) => ({
  loadInitialState: () => {
    dispatch(loadInitialState());
  },
  signIn: (provider, username, password) => {
    dispatch(signIn({ provider, username, password }));
  },
  signUp: (provider, username, email, password, accept) => {
    dispatch(signUp({ provider, username, email, password, accept }));
  },
  lostPassword: (provider, email) => {
    dispatch(lostPassword({ provider, email }));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(AuthenticateBase);
