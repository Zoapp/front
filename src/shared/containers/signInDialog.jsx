/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Zrmc, {
  Dialog,
  DialogFooter,
  DialogBody,
  FormField,
  TextField,
  Button,
} from "zrmc";
import { signIn } from "../actions/auth";

export class SignInDialogBase extends Component {
  state = {
    username: "",
    password: "",
  };

  handleCloseDialog = () => {
    Zrmc.closeDialog();
    if (this.props.onClosed !== null) {
      this.props.onClosed();
    }
  };

  handleSignIn = (e) => {
    e.preventDefault();
    const { provider } = this.props;
    const { username, password } = this.state;

    if (username !== "" && password !== "") {
      this.props.signIn(provider, username, password);
      this.handleCloseDialog();
    }
  };

  createChangeHandler = (field) => (e) => {
    this.setState({ [field]: e.target.value });
  };

  render() {
    const { username, password } = this.state;

    return (
      <form id="signin-dialog-form" onSubmit={this.handleSignIn}>
        <Dialog
          id={this.props.id}
          onClose={this.handleCloseDialog}
          header="Your credentials"
          width="320px"
        >
          <DialogBody>
            <FormField style={{ display: "block" }}>
              <TextField
                id="signin-dialog-username"
                defaultValue={username}
                onChange={this.createChangeHandler("username")}
                label="Username | Email"
                style={{ width: "200px" }}
                autoComplete="username email"
                required
              />
            </FormField>
            <FormField style={{ display: "block" }}>
              <TextField
                id="signin-dialog-password"
                defaultValue={password}
                onChange={this.createChangeHandler("password")}
                label="Password"
                type="password"
                style={{ width: "200px" }}
                autoComplete="password"
                required
              />
            </FormField>
          </DialogBody>
          <DialogFooter>
            <Button
              id="signin-dialog-cancel-button"
              type="button"
              onClick={(e) => {
                e.preventDefault();
                this.handleCloseDialog();
              }}
            >
              Cancel
            </Button>
            <Button type="submit">Sign in</Button>
          </DialogFooter>
        </Dialog>
      </form>
    );
  }
}

SignInDialogBase.defaultProps = {
  id: null,
  onClosed: null,
  provider: null,
};

SignInDialogBase.propTypes = {
  id: PropTypes.string,
  onClosed: PropTypes.func,
  provider: PropTypes.string,
  signIn: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  signIn: (provider, username, password) => {
    dispatch(signIn({ provider, username, password }));
  },
});

export default connect(
  null,
  mapDispatchToProps,
)(SignInDialogBase);
