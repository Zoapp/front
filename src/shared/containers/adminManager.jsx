/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import Zrmc, { Grid, Inner, Cell, Button, Content, TextField } from "zrmc";
import { connect } from "react-redux";
import { TableComponent } from "zoapp-ui";
import { appSetTitle } from "../actions/app";
import { apiSetAdminParametersRequest } from "../actions/api";

import Loading from "../components/loading";
import SignInForm from "./signInForm";

// import PluginsManager from "../utils/pluginsManager";

const infoStyleD = {
  fontSize: "16px",
  fontWeight: "400",
  color: "#666",
  padding: "24px",
  lineHeight: "1.1",
};

class AdminManager extends Component {
  static onActionTunnel() {
    // console.log("onActionTunnel", dialog, action);
    Zrmc.closeDialog();
  }

  constructor(props) {
    super(props);
    // const pluginsManager = new PluginsManager();
    this.state = {
      // pluginsManager,
      tunnelParams: null,
      backendParams: null,
      emailServerParams: null,
    };
    this.props.appSetTitle("Admin");
  }

  onChangeTunnel = (tunnelParams) => {
    this.setState({ tunnelParams });
  };

  onSaveBackend() {
    if (this.state.tunnelParams) {
      // WIP save tunnel parameters
      const tunnel = this.state.tunnelParams.active || "None";
      this.setState({ tunnelParams: null });
      this.props.apiSetAdminParametersRequest({ tunnel });
    }
  }

  displayTunnelDialog() {
    this.todo = {};
    // const { params } = this.props.admin;
    // const backend = this.state.backendParams || params.backend || { };
    // const tunnelParams = this.state.tunnelParams || backend.tunnel || { };
    /* const content = <div>Tunnel</div>;
    Zrmc.showDialog({
      title: "Tunnel settings",
      content,
      width: "520px",
      onAction: AdminManager.onActionTunnel,
    }); */
  }

  render() {
    let { isLoading } = this.props;
    if (!isLoading && !this.props.admin && this.props.isSignedIn) {
      isLoading = true;
    }
    if (!this.props.isSignedIn) {
      return <SignInForm />;
    } else if (isLoading || this.props.admin == null) {
      return <Loading />;
    }
    const active = this.props.activeTab;
    let content = "";
    if (active === 0) {
      const saveDisabled = !this.state.botParams;
      content = (
        <Grid>
          <Cell
            className="mdl-color--white"
            col={12}
            style={{ display: "table" }}
          >
            <div style={{ width: "200px", display: "table-cell" }}>
              <div
                style={{
                  position: "absolute",
                  width: "180px",
                  height: "180px",
                  margin: "24px",
                  backgroundColor: "#ddd",
                }}
              />
            </div>
            <div style={{ display: "table-cell" }}>TODO</div>
            <div>
              <Button
                raised
                colored
                disabled={saveDisabled}
                style={{ float: "right", margin: "24px" }}
              >
                SAVE
              </Button>
            </div>
          </Cell>
        </Grid>
      );
    } else if (active === 1) {
      content = (
        <Grid>
          <div>Services</div>
        </Grid>
      );
    } else if (active === 2) {
      const items = [];
      const status = "you";
      const { user, profile } = this.props;
      // const spanStyle = { width: "160px" };
      const values = [];
      values.push(profile.username);
      values.push(profile.email);
      values.push(user.attributes.scope);
      values.push(status);
      items.push({ id: 1, values, icon: `../images/${profile.avatar}.png` });
      const headers = ["", "username", "email", "role", "status"];
      const title = (
        <div style={infoStyleD}>
          You could give an access to your collaborators here.
          <Button
            raised
            colored
            style={{ float: "right", marginBottom: "24px" }}
          >
            ADD
          </Button>
        </div>
      );
      content = (
        <Grid>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              <div>
                <TableComponent
                  title={title}
                  headers={headers}
                  items={items}
                  selectedItem={-1}
                  onSelect={() => {}}
                />
              </div>
            </Cell>
          </Inner>
        </Grid>
      );
    } else if (active === 3) {
      const { params } = this.props.admin;
      const emailServer = this.state.emailParams || params.emailServer || {};
      const backend = this.state.backendParams || params.backend || {};
      // const tunnelParams = this.state.tunnelParams || backend.tunnel || {};
      // const hasTunnelParams = !!this.state.tunnelParams;
      const saveBackendDisabled = !(
        this.state.backendParams || this.state.tunnelParams
      );
      const saveEmailDisabled = !this.state.emailServerParams;
      content = (
        <Grid>
          <Cell className="mdl-color--white" col={12}>
            <div style={infoStyleD}>
              Backend configuration
              <Button
                raised
                colored
                disabled={saveBackendDisabled}
                style={{ float: "right" }}
                onClick={(e) => {
                  e.preventDefault();
                  this.onSaveBackend();
                }}
              >
                SAVE
              </Button>
            </div>
            <form style={infoStyleD}>
              <div style={{ width: "520px" }}>
                <TextField
                  onChange={() => {}}
                  label="Public Api url"
                  style={{ width: "400px" }}
                  value={backend.publicUrl}
                />
              </div>
              <div>
                <TextField
                  onChange={() => {}}
                  label="Api url"
                  disabled
                  style={{ width: "400px" }}
                  value={backend.apiUrl}
                />
              </div>
              <div>
                <TextField
                  onChange={() => {}}
                  label="Auth url"
                  disabled
                  style={{ width: "400px" }}
                  value={backend.authUrl}
                />
              </div>
              <div>
                <TextField
                  onChange={() => {}}
                  label="AppId"
                  disabled
                  style={{ width: "400px" }}
                  value={backend.clientId}
                />
              </div>
              <div>
                <TextField
                  onChange={() => {}}
                  label="Secret"
                  disabled
                  style={{ width: "400px" }}
                  value={backend.clientSecret}
                />
              </div>
            </form>
            <div />
          </Cell>
          <Cell className="mdl-color--white" col={12}>
            <div style={infoStyleD}>
              Email server configuration<Button
                raised
                colored
                disabled={saveEmailDisabled}
                style={{ float: "right" }}
              >
                SAVE
              </Button>
            </div>
            <form style={infoStyleD} autoComplete="nope">
              <div>
                <TextField
                  onChange={() => {}}
                  label="Server address"
                  style={{ width: "400px" }}
                  value={emailServer.url}
                />
              </div>
              <div>
                <TextField
                  onChange={() => {}}
                  label="Username"
                  autoComplete="new-password"
                  style={{ width: "400px" }}
                  value={emailServer.username}
                />
              </div>
              <div>
                <TextField
                  onChange={() => {}}
                  label="Password"
                  autoComplete="new-password"
                  type="password"
                  style={{ width: "400px" }}
                  value={emailServer.password}
                />
              </div>
            </form>
            <div />
          </Cell>
          <Cell className="mdl-color--white" col={12}>
            <div style={infoStyleD}>
              <span style={{ color: "#d50000" }}>Delete this assistant</span>
              <Button
                raised
                colored
                style={{
                  float: "right",
                  marginBottom: "24px",
                  backgroundColor: "#d50000",
                }}
              >
                DELETE
              </Button>
            </div>
            <div />
          </Cell>
        </Grid>
      );
    }
    return (
      <Content className="mdl-color--grey-100">
        <section>{content}</section>
      </Content>
    );
  }
}

AdminManager.defaultProps = {
  admin: null,
  isLoading: false,
  isSignedIn: false,
  user: null,
  profile: {},
  activeTab: 0,
};

AdminManager.propTypes = {
  admin: PropTypes.shape({ params: PropTypes.shape({}).isRequired }),
  isLoading: PropTypes.bool,
  isSignedIn: PropTypes.bool,
  user: PropTypes.shape({}),
  activeTab: PropTypes.number,
  appSetTitle: PropTypes.func.isRequired,
  apiSetAdminParametersRequest: PropTypes.func.isRequired,
  profile: PropTypes.shape({}),
};

AdminManager.contextTypes = {
  activeTab: PropTypes.number,
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading;
  const { user } = state;
  const profile = user ? user.profile : null;
  return {
    admin,
    isLoading,
    isSignedIn,
    user,
    profile,
  };
};

const mapDispatchToProps = (dispatch) => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
  apiSetAdminParametersRequest: (params) => {
    dispatch(apiSetAdminParametersRequest(params));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(AdminManager);
