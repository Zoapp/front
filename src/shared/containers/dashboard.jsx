/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Grid, Inner, Cell } from "react-material-cw";
import Loading from "../components/loading";
import { appSetTitle } from "../actions/app";

class Dashboard extends Component {
  componentWillMount() {
    this.props.appSetTitle("Dashboard");
  }

  render() {
    const { isLoading } = this.props;
    if (!this.props.isSignedIn) {
      return (<div>You need to sign in...</div>);
    } else if (isLoading || this.props.admin == null) {
      return (<Loading />);
    }
    // const { admin } = this.props;
    return (
      <div className="mdl-color--grey-100">
        <Grid>
          <Inner>
            <Cell className="mdl-color--white" span={12}>
              Dashboard
            </Cell>
          </Inner>
        </Grid>
      </div>
    );
  }
}

Dashboard.defaultProps = {
  admin: null,
  isLoading: false,
  isSignedIn: false,
};

Dashboard.propTypes = {
  admin: PropTypes.shape({ params: PropTypes.shape({}).isRequired }),
  isLoading: PropTypes.bool,
  isSignedIn: PropTypes.bool,
  appSetTitle: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading;
  return { admin, isLoading, isSignedIn };
};

const mapDispatchToProps = dispatch => ({
  appSetTitle: (titleName) => {
    dispatch(appSetTitle(titleName));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
