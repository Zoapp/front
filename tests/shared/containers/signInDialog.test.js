/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import { createFakeEvent } from "tests/helpers";
import { SignInDialogBase } from "shared/containers/signInDialog";

describe("containers/signInDialog", () => {
  it("renders correctly", () => {
    const signInSpy = jest.fn();
    const component = renderer.create(<SignInDialogBase signIn={signInSpy} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("sign in with fullfilled form", () => {
    const signInSpy = jest.fn();
    const wrapper = shallow(<SignInDialogBase signIn={signInSpy} />);

    wrapper
      .find("#signin-dialog-username")
      .simulate("change", { target: { value: "username" } });
    expect(wrapper.state("username")).toEqual("username");

    wrapper
      .find("#signin-dialog-password")
      .simulate("change", { target: { value: "password" } });
    expect(wrapper.state("password")).toEqual("password");

    wrapper.find("#signin-dialog-form").simulate("submit", createFakeEvent());

    expect(signInSpy).toHaveBeenCalled();
  });
});
