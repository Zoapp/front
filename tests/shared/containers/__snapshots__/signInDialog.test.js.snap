// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/signInDialog renders correctly 1`] = `
<form
  id="signin-dialog-form"
  onSubmit={[Function]}
>
  <dialog
    className="rmdc-dialog mdc-dialog mdc-dialog--open"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="presentation"
  >
    <div
      className="mdc-dialog__surface"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="presentation"
      style={
        Object {
          "minWidth": "320px",
          "width": "320px",
        }
      }
    >
      <header
        className="mdc-dialog__header"
      >
        Your credentials
      </header>
      <section
        className="mdc-dialog__body"
      >
        <section
          className="mdc-dialog__body"
        >
          <div
            className="mdc-form-field"
            style={
              Object {
                "display": "block",
              }
            }
          >
            <div
              className="mdc-text-field mdc-text-field--upgraded"
              style={
                Object {
                  "width": "200px",
                }
              }
            >
              <input
                autoComplete="username email"
                className="mdc-text-field__input"
                defaultValue=""
                id="signin-dialog-username"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              <label
                className="mdc-floating-label"
                focused="false"
                htmlFor="signin-dialog-username"
                style={Object {}}
              >
                Username | Email
              </label>
              <div
                className="mdc-text-field__bottom-line"
              />
            </div>
          </div>
          <div
            className="mdc-form-field"
            style={
              Object {
                "display": "block",
              }
            }
          >
            <div
              className="mdc-text-field mdc-text-field--upgraded"
              style={
                Object {
                  "width": "200px",
                }
              }
            >
              <input
                autoComplete="password"
                className="mdc-text-field__input"
                defaultValue=""
                id="signin-dialog-password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
              />
              <label
                className="mdc-floating-label"
                focused="false"
                htmlFor="signin-dialog-password"
                style={Object {}}
              >
                Password
              </label>
              <div
                className="mdc-text-field__bottom-line"
              />
            </div>
          </div>
        </section>
      </section>
      <footer
        className="mdc-dialog__footer"
      >
        <button
          className="mdc-dialog__footer__button mdc-dialog__footer__button--cancel mdc-button"
          onClick={[Function]}
        >
          
          Cancel
        </button>
        <button
          className="mdc-dialog__footer__button mdc-dialog__footer__button--cancel mdc-button"
          onClick={null}
        >
          
          Sign in
        </button>
      </footer>
    </div>
  </dialog>
</form>
`;
